File: Searching/01-linear-search.cpp
#include <bits/stdc++.h>
using namespace std;

int linear_search(const vector<int> &arr, int x)
{
    int len = arr.size();
    for (int i = 0; i < len; i++)
    {
        if (arr[i] == x)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    vector<int> arr = {1, 23, 13, 22, 55, 29};
    cout << linear_search(arr, 13) << '\n';
}

File: days/05_search/ts/01_linear_search.ts
const linear_search = (a: number[], t: number): number => {
    const l = a.length;
    for (let i = 0; i < l; i++) {
        if (t === a[i]) {
            return i;
        }
    }
    return -1;
};

const a = [1, 23, 22, 13, 55, 29];
console.log(linear_search(a, 13));
